### PR TITLE
Verify spark compatibility in native-fs special chars test

### PR DESCRIPTION
**Pull request by anusudarsan**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java [hunk 1]
```diff
@@ -121,13 +121,21 @@ protected void testPathContainsSpecialCharacter(String schemaLocation, String pa
                     row(11, "with%%percents"),
                     row(12, "with space"));
             assertThat(onTrino().executeQuery(format("SELECT * FROM %1$s.test.%2$s", getCatalogName(), tableName))).containsOnly(expectedRows);
+            if (testSparkCompatibility()) {
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java:124]: Can we leave a code comment why this condition is required? It's not clear when reading the code.

Comment (anusudarsan) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java:124]: ya will do. the delta container is not configured in this env for spark. We will have to either create a separate env for testing delta or have a consolidated image for spark iceberg and delta

**Comment by ebyhr** [comment]
/test-with-secrets sha=a15549452071b337de6e457d659010bb05b464a2

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java (hunk 1 above)
Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java]: Thanks for your explanation. I get it now. Then, how about using the existing `getCatalogName` method?

**Review by ebyhr — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java (hunk 1 above)
Comment (anusudarsan) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java]: done

Comment (findinpath) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/BaseTestTableFormats.java]: > I guess there are tests already for spark compat in delta-oss , so not a big deal if we skip

We need compatibility tests between Trino and Spark on Azure & GCS to ensure that what we write is actually readable, and hopefully correct,  by Spark.

>  the delta container is not configured in this env for spark. 

@anusudarsan  Can you please add a follow-up to address this point?

**Pull request merged**